### PR TITLE
[CELEBORN-1595] Support quota low watermark for checking quota available

### PR DESCRIPTION
**Pull request by s0nskar**
### What changes were proposed in this pull request?

Support quota low watermark for checking quota available. This will not allow new jobs to run on Celeborn if quota used is above lowWatermark.

### Why are the changes needed?

Currently we allow jobs to run even if we're just about to breach quota limits. This is not ideal behaviour, ideally we should not allow any new jobs to run on Celeborn after certain threshold (called lowWatermark here). This will ensure current running jobs will use the quota and will not go way beyond quota usage. 

I'll also follow up with a PR to throw CelebornIOException, if quota is breached.

### Does this PR introduce _any_ user-facing change?

NA

### How was this patch tested?

UTs

### CODE REVIEW

**Comment by s0nskar** [comment]
IMO this configuration can be a dynamic configuration as well. Wdyt?

**Comment by s0nskar** [comment]
cc: @SteNicholas @waitinfuture @mridulm

**Comment by s0nskar** [comment]
Also, IMO quota should be a hard limit any application breaching quota should get a `CelebornIOException`. We can improve the application heartbeat to check for current quota.

**Comment by s0nskar** [comment]
ping @SteNicholas @waitinfuture

**Comment by s0nskar** [comment]
ping @FMX @SteNicholas @waitinfuture

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala [hunk 1]
```diff
@@ -39,10 +39,23 @@ class QuotaManager(celebornConf: CelebornConf, configService: ConfigService) ext
...
   }
 
-  def checkQuotaSpaceAvailable(
+  private def getQuotaWithWatermark(
       userIdentifier: UserIdentifier,
-      resourceResumption: ResourceConsumption): (Boolean, String) = {
+      quotaWatermark: Double): Quota = {
     val quota = getQuota(userIdentifier)
+    Quota(
+      (quota.diskBytesWritten * quotaWatermark).toLong,
```
Comment (waitinfuture) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala:47]: Hi @s0nskar , I don't quite understand why can't we just config the quota to `quota.diskBytesWritten * 1.0 (some user defined factor)`? Is it necessary to introduce this new config?

Comment (s0nskar) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala:47]: @waitinfuture Currently Celeborn consider quota as soft limits and just stop allowing new jobs on use Celeborn if the quota is breached while the existing jobs keeps on running while breaching the quota. IMO there are two shortcoming in current quota implementation –

- It allows new jobs to onboard even if quota is just about to be breached. Ideally it should stop onboarding jobs after a certain threshold to not go overboard with the quota limits (This PR addresses that)
- Killing jobs if quota for an user is breached (I was planning to implement this but noticed that @leixm raised a PR to handle this today - https://github.com/apache/celeborn/pull/2801)

**Pull request closed without merging**